tests/functional/setups/manu_sneddon_2d.py

```diff
@@ -0,0 +1,436 @@
+        
+        
+        u_a = -analytical_displacements(self.a, eta, self.p0, self.shear_modulus, self.poi)
+        u_bc = assign_bem(sd, h / 2, box_faces, self.theta, bem_centers, u_a, self.poi)
+        return u_bc
+    
+
+    def exact_sol_fracture( self,
+    gb: pp.MixedDimensionalGrid) -> tuple:
+        """
+        Compute Sneddon's analytical solution for the pressurized crack
+        problem in question.
+
+        Source: Sneddon Fourier transforms 1951 page 425 eq 92
+
+        Parameter
+            gb: Grid object
+            a: Half fracture length of fracture
+            eta: Distance from fracture centre
+            p0: Internal constant pressure inside the fracture
+            G: Shear modulus
+            poi: Poisson ratio
+            height,length: Height and length of domain
+
+        Return:
+            A tuple containing two vectors: a list of distances from the fracture center to each fracture coordinate, and the corresponding analytical apertures.
+        """
+        ambient_dim = gb.dim_max()
+        g_1 = gb.subdomains(dim=ambient_dim - 1)[0]
+        fracture_center = np.array([self.length / 2, self.height / 2, 0])
+
+        fracture_faces = g_1.cell_centers
+
+        # compute distances from fracture centre with its corresponding apertures
+        eta = compute_eta(fracture_faces, fracture_center)
+        apertures = analytical_displacements(self.a, eta, self.p0, self.shear_modulus, self.poi)
+
+        return eta, apertures
+
+
+
+
+
+class ModifiedGeometry(PorePyModel):
+    def set_domain(self) -> None:
+        """Defining a two-dimensional square domain with sidelengths length and height."""
+
+        self._domain = pp.Domain(
+            bounding_box={
+                "xmin": 0,
+                "ymin": 0,
+                "xmax": self.params["length"],
+                "ymax": self.params["height"],
+            }
+        )
+
+    def grid_type(self) -> Literal["simplex", "cartesian", "tensor_grid"]:
+        """Choosing the grid type for our domain."""
+        return self.params.get("grid_type", "simplex")
+
+    def set_fractures(self):
+        """Setting the fractures in the domain from params."""
+
+        points = self.params["frac_pts"]
+        self._fractures = [pp.LineFracture(points)]
+
+
+class ModifiedBoundaryConditions(PorePyModel):
+    def bc_type_mechanics(self, sd: pp.Grid) -> pp.BoundaryConditionVectorial:
+        """Set boundary condition type for the problem.
+        Parameters:
+            sd: Subdomain for which the boundary conditions are to be set.
+        Returns:
+            Boundary condition type for the problem.
+        """
+        bounds = self.domain_boundary_sides(sd)
+
+        # Set the type of west and east boundaries to Dirichlet. North and south are
+        # Neumann by default.
+        bc = pp.BoundaryConditionVectorial(sd, bounds.all_bf, "dir")
+        frac_face = sd.tags["fracture_faces"]
```

Comment:
I think there's a function for this. internal_bc_to_dirichlet or whatever.

Reply:
I also believe this to be the default behaviour - all Dirichlet on external and internal boundaries.

Reply:
Changed these lines to `internal_to_dirichlet`